include/exadg/postprocessor/pointwise_output_generator_base.cpp

```diff
@@ -136,9 +102,10 @@ PointwiseOutputGeneratorBase<dim, Number>::setup_base(
     componentwise_result.reinit(pointwise_output_data.evaluation_points.size());
 
     // number of samples to write into file
-    n_out_samples = 1 + static_cast<unsigned int>(std::ceil(
-                          (pointwise_output_data.end_time - pointwise_output_data.start_time) /
-                          pointwise_output_data.interval_time));
+    n_out_samples = 1 + static_cast<unsigned int>(
+                          std::ceil((pointwise_output_data.time_control_data.end_time -
+                                     pointwise_output_data.time_control_data.start_time) /
+                                    pointwise_output_data.time_control_data.trigger_interval));
```

Comment:
This looks like `trigger_every_time_steps` of `TimeControl` cannot be used for this module? I think this calls for a special solution for this particular postprocessing module.

Reply:
The module uses TimeControl. We just have to know before writing how much memory we need to allocate. Therefore, I use the data that is in TimeControl. You are right, this is only implemented for time intervals. This is because we do not know how much memory to allocate for the time step case and adaptive timestepping.

If it is OK I will add an assert and keep the rest

Reply:
Yes, that good. I think an assert will avoid errors.

Reply:
[68b7a04](https://github.com/exadg/exadg/pull/262/commits/68b7a047785a84d53e2c2a7ef610e44ad09c7993)